src/interactivity/win32/Clipboard.cpp

```diff
@@ -59,29 +59,60 @@ void Clipboard::Paste()
         return;
     }
 
-    const auto handle = GetClipboardData(CF_UNICODETEXT);
-    if (!handle)
+    // This handles most cases of pasting text as the OS converts most formats to CF_UNICODETEXT automatically.
+    if (const auto handle = GetClipboardData(CF_UNICODETEXT))
     {
-        return;
-    }
+        const wil::unique_hglobal_locked lock{ handle };
+        const auto str = static_cast<const wchar_t*>(lock.get());
+        if (!str)
+        {
+            return;
+        }
 
-    // Clear any selection or scrolling that may be active.
-    Selection::Instance().ClearSelection();
-    Scrolling::s_ClearScroll();
+        // Clear any selection or scrolling that may be active.
+        Selection::Instance().ClearSelection();
+        Scrolling::s_ClearScroll();

```

Comment:
this is common with the other branch through `StringPaste` below; is there any need (value?) in deduplicating?

Reply:
Hmm, let me try to restructure the code a little bit and try to unify it with the other HDROP handler in conhost.